crates/client/src/client/memoize_client_handle.rs

```diff
@@ -29,7 +30,7 @@ use crate::op::Query;
 #[must_use = "queries can only be sent through a ClientHandle"]
 pub struct MemoizeClientHandle<H: ClientHandle> {
     client: H,
-    active_queries: Arc<Mutex<HashMap<Query, RcFuture<<H as DnsHandle>::Response>>>>,
+    active_queries: Arc<Mutex<HashMap<Query, RcStream<<H as DnsHandle>::Response>>>>,
```

Comment:
I ran into this one when working on my own version of this change. I'm not sure `MemoizeClientHandle` makes sense in the context of a stream, as we'd need to store all records for the entire stream...